src/server/game/Spells/Spell.cpp

```diff
@@ -2322,7 +2322,8 @@ void Spell::TargetInfo::PreprocessTarget(Spell* spell)
     if (_spellHitTarget)
     {
         // if target is flagged for pvp also flag caster if a player
-        if (unit->IsPvP() && spell->m_caster->GetTypeId() == TYPEID_PLAYER)
+        // but respect current pvp rules (buffing/healing npcs flagged for pvp only flags you if they are in combat)
+        if (unit->IsPvP() && (unit->IsInCombat() || unit->IsCharmedOwnedByPlayerOrPlayer()) && spell->m_caster->GetTypeId() == TYPEID_PLAYER)
```

Comment:
```suggestion
        if (unit->IsPvP() && (unit->IsInCombat() || unit->IsControlledByPlayer()) && spell->m_caster->GetTypeId() == TYPEID_PLAYER)
```

Attacking a player who is mind controlled/charmed by a creature should not set pvp flag